src/lib/types.ts

```diff
@@ -80,18 +82,36 @@ export abstract class PostgrestBuilder<T> implements PromiseLike<PostgrestRespon
       body: JSON.stringify(this.body),
     })
       .then(async (res) => {
-        let error, data
+        let error, data, count
         if (res.ok) {
           error = null
-          const isReturnMinimal = this.headers['Prefer']?.split(',').includes('return=minimal')
-          data = isReturnMinimal ? null : await res.json()
+          if (this.method !== 'HEAD') {
+            const isReturnMinimal = this.headers['Prefer']?.split(',').includes('return=minimal')
+            data = isReturnMinimal ? null : await res.json()
+          } else {
+            data = null
+          }
+
+          const countHeader = this.headers['Prefer']?.match(/count=(exact|planned|estimated)/)
```

Comment:
I feel like there is a better way of extracting the `count` value from the header here. Any suggestions would be great!

Reply:
I think this is fine for now. An alternative would be to add a `prefers` object property to `PostgrestBuilder` and join everything in `then()`.

Reply:
@soedirgo  Okay, I will leave it as it is for now. Thanks for checking!